src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -94,6 +95,14 @@ public TimeSpan UserProcessorTime
             }
         }
 
+        public int HandleCount
+        {
+            get
+            {
+                return Directory.GetDirectories("/dev/fd", "*", SearchOption.TopDirectoryOnly).Length;
```

Comment:
Doesn't this end up returning the number of descriptors for the current process, which may not be the process represented by Id / this Process object?